src/towncrier/_builder.py

```diff
@@ -106,10 +107,22 @@ def __call__(self, section_directory: str = "") -> str:
 def find_fragments(
     base_directory: str,
     config: Config,
+    strict: bool | None,
 ) -> tuple[Mapping[str, Mapping[tuple[str, str, int], str]], list[tuple[str, str]]]:
     """
     Sections are a dictonary of section names to paths.
+
+    In strict mode, raise ClickException if any fragments have an invalid name.
```

Comment:
I know that this questions might be easy to answer in the context of this PR, but I think that the documentation should be improved... so that it will make sense when read outside of this PR.


Is this exception raised only in `strict mode` ?
What is `strict mode` ?

--------

What about raising the exception when `build_ignore_filenames` is set ?